File: Epic_Final_Project-main/Epic_Final_Project-main/Our E_F_C_P_E_F_ E Final Project/Models/PotionAndWiz.cs
using Dapper.Contrib.Extensions;
using MySql.Data.MySqlClient;

namespace Our_E_F_C_P_E_F__E_Final_Project
{
    [Table("PotionAndWiz")]
    public class PotionAndWiz
    {
        public string id { get; set; }
        public string potion_name { get; set; }
        public string? photoUrl { get; set; }
        public string effects { get; set; }
        public string? sideEffects { get; set; }
        public string? characteristics { get; set; }
        public int price { get; set; }
        public bool inStock { get; set; }
        //public string?[] ingredient_id { get; set; }
        public string createdBy { get; set; }

        public string wizard_id { get; set; }
        public string wizard_first_name { get; set; }
        public string wizard_last_name { get; set; }
        public string? wizard_picture { get; set; }

        public static MySqlConnection DB;
        public static List<PotionAndWiz> GetAllInfo()
        {
            DB.Open();
            var result = DB.GetAll<PotionAndWiz>().ToList();
            DB.Close();

            return result;
        }
        public static PotionAndWiz GetOneInfo(string id)
        {
            DB.Open();
            var result = DB.Get<PotionAndWiz>(id);
            DB.Close();

            return result;
        }
        
    }
}

File: potionshopFront/src/app/potionandwiz.ts
export interface PotionAndWiz {
    id: string;
    potion_name: string;
    photoUrl: string;
    effects: string;
    sideEffects: string;
    characteristics: string;
    price: number;
    inStock: boolean;
    createdBy: string;
    wizard_id: string;
    wizard_first_name: string;
    wizard_last_name: string;
    wizard_picture: string;

};

export interface Potion {
id:string
potion_name: string;
photoUrl: string;
effects?: string|null;
sideEffects?: string|null;
characteristics?: string|null;
price: number;
inStock: boolean;
createdBy: string;
};

export interface Wizard{
    wizard_id: string;
    wizard_first_name: string;
    wizard_last_name: string;
    wizard_picture: string;
};
